yt/yt/server/queue_agent/queue_controller.cpp

```diff
@@ -1144,23 +1196,60 @@ class TOrderedDynamicTableController
             }
         }
 
-        //! Updates partition contexts in accordance with the offsets of vital consumers.
+        //! Updates partition contexts in accordance with the offsets of vital consumers and exports.
         //! Only affects the minimum trimmed row count.
-        void HandleVitalConsumers()
+        void HandleVitalConsumersAndExports()
         {
+            auto handleQueueExportByPartitionAndReturnMinTrimmedRowCount = [this] (TPartitionTrimContext& partitionContext) -> std::optional<i64> {
+                if (QueueExportProgress.empty()) {
+                    return {};
+                }
+
+                std::optional<i64> minTrimmedRowCount;
+                for (auto& [exportName, exportProgress] : QueueExportProgress) {
+                    auto partitionProgressIt = exportProgress->Tablets.find(partitionContext.PartitionIndex);
+                    if (partitionProgressIt == exportProgress->Tablets.end()) {
+                        partitionContext.Update({
+                                .PartitionError = TError(
+                                    "Not trimming partition %v since export %v is missing this partition",
+                                    partitionContext.PartitionIndex,
+                                    exportName),
+                                .IsErrorCritical = true,
+                            });
+                        break;
+                    }
+                    const auto& partitionProgress = partitionProgressIt->second;
+                    minTrimmedRowCount = MinOrValue<i64>(
+                        minTrimmedRowCount,
+                        partitionProgress->RowCount);
+                }
+                return minTrimmedRowCount;
+            };
+
             for (auto& partitionContext : Context.Partitions) {
                 if (!partitionContext) {
                     continue;
                 }
 
                 std::optional<i64> minTrimmedRowCount;
+
+                // Handle vital consumers.
                 for (const auto& [consumerRef, consumerSubSnapshot] : VitalConsumerSubSnapshots) {
                     minTrimmedRowCount = MinOrValue<i64>(
                         minTrimmedRowCount,
                         // NextRowIndex should always be present in the snapshot.
                         consumerSubSnapshot->PartitionSnapshots[partitionContext.PartitionIndex]->NextRowIndex);
                 }
 
+                // Handle queue exports.
+                minTrimmedRowCount = MinOrValue<i64>(
+                    minTrimmedRowCount,
+                    handleQueueExportByPartitionAndReturnMinTrimmedRowCount(partitionContext));
+
+                if (partitionContext.HasCriticalError()) {
```

Comment:
I don't think you need this now and especially when you address the other comments.
The code later on won't actually do anything if there is an error.